kensu/json/__init__.py

```diff
@@ -0,0 +1,70 @@
+import json as js
+from json import *
+
+from kensu.boto3 import ksu_dict
+from kensu.botocore.response import ksu_bytes
+from kensu.client import DataSourcePK, DataSource, FieldDef, SchemaPK, Schema
+from kensu.itertools import kensu_list
+from kensu.utils.helpers import logical_naming_batch
+from kensu.utils.kensu_provider import KensuProvider
+
+
+def wrap_loads(method):
+    def wrapper(*args, **kwargs):
+
+        result = method(*args, **kwargs)
+
+        kensu = KensuProvider().instance()
+
+        data = args[0]
+        if isinstance(result,dict):
+            result = ksu_dict(result)
+        elif isinstance(result,list):
+            result = kensu_list(result)
+        if isinstance(data,ksu_bytes):
+            result.ksu_metadata = data.ksu_metadata
+
+            origin_location = result.ksu_metadata['origin_location']
+            origin_name = result.ksu_metadata['origin_name']
+
+            if kensu.logical_naming == 'ReplaceNumbers':
+                logical = logical_naming_batch(origin_name)
+            else:
+                logical = origin_name
+
+            #FIXME we should put this in extractors
+            result_pk = DataSourcePK(location=origin_location,
+                                     physical_location_ref=kensu.default_physical_location_ref)
+            result_ds = DataSource(name=origin_name, categories=['logical::' + logical], format=origin_name.split('.')[-1],
+                                   pk=result_pk)._report()
+
+            fields_set = set()
+
+            if isinstance(result, list):
+                for element in result:
+                    for e in element.keys():
+                        fields_set.add(('[].' + str(e), type(e).__name__))
```

Comment:
P.S. if list/dict has many unique values at the first level it would report too many of them...